src/engine/TransitivePathImpl.h

```diff
@@ -322,26 +309,49 @@ class TransitivePathImpl : public TransitivePathBase {
   static cppcoro::generator<TableColumnWithVocab> setupNodes(
       const TransitivePathSide& startSide,
       std::shared_ptr<const Result> startSideResult) {
+    ColumnIndex joinColumn = startSide.treeAndCol_.value().second;
+    size_t cols = startSide.treeAndCol_.value().first->getResultWidth();
+    std::vector<ColumnIndex> columnsWithoutJoinColumn =
+        computeColumnsWithoutJoinColumn(joinColumn, cols);
     if (startSideResult->isFullyMaterialized()) {
       // Bound -> var|id
-      ql::span<const Id> startNodes = startSideResult->idTable().getColumn(
-          startSide.treeAndCol_.value().second);
-      co_yield TableColumnWithVocab{&startSideResult->idTable(), startNodes,
-                                    startSideResult->getCopyOfLocalVocab()};
+      ql::span<const Id> startNodes =
+          startSideResult->idTable().getColumn(joinColumn);
+      co_yield TableColumnWithVocab{
+          startSideResult->idTable().asColumnSubsetView(
+              columnsWithoutJoinColumn),
+          startNodes, startSideResult->getCopyOfLocalVocab()};
     } else {
       for (auto& [idTable, localVocab] : startSideResult->idTables()) {
         // Bound -> var|id
-        ql::span<const Id> startNodes =
-            idTable.getColumn(startSide.treeAndCol_.value().second);
-        co_yield TableColumnWithVocab{&idTable, startNodes,
-                                      std::move(localVocab)};
+        ql::span<const Id> startNodes = idTable.getColumn(joinColumn);
+        co_yield TableColumnWithVocab{
+            idTable.asColumnSubsetView(columnsWithoutJoinColumn), startNodes,
+            std::move(localVocab)};
       }
     }
   }
 
   virtual T setupEdgesMap(const IdTable& dynSub,
                           const TransitivePathSide& startSide,
                           const TransitivePathSide& targetSide) const = 0;
+
+ private:
+  // Helper function to filter the join column to not add it twice to the
+  // result.
+  static std::vector<ColumnIndex> computeColumnsWithoutJoinColumn(
+      ColumnIndex joinColumn, size_t totalColumns) {
+    std::vector<ColumnIndex> columnsWithoutJoinColumn;
+    AD_CORRECTNESS_CHECK(totalColumns > 0);
+    columnsWithoutJoinColumn.reserve(totalColumns - 1);
+    for (ColumnIndex i = 0; i < totalColumns; ++i) {
+      if (i == joinColumn) {
+        continue;
+      }
+      columnsWithoutJoinColumn.push_back(i);
+    }
+    return columnsWithoutJoinColumn;
```

Comment:
views::iota | filter != joinColumn | toVector;
(toVector is `range::copy(...., back_inserter{...})_;)